amcl/src/amcl_node.cpp

```diff
@@ -685,82 +745,135 @@ void AmclNode::runFromBag(const std::string &in_bag_fn)
+                                  last_published_pose.pose.covariance[6*0+0]),
+       rclcpp::parameter::ParameterVariant("initial_cov_yy",
+                                  last_published_pose.pose.covariance[6*1+1]),
+       rclcpp::parameter::ParameterVariant("initial_cov_yy",
+                                  last_published_pose.pose.covariance[6*5+5]),
+      }
+    );
+  if (!set_parameters_results.successful) {
+    fprintf(stderr, "Failed to set parameter: %s\n",
+            set_parameters_results.reason.c_str());
+  }
 }
 
 void AmclNode::updatePoseFromServer()
 {
-  init_pose_[0] = 0.0;
-  init_pose_[1] = 0.0;
-  init_pose_[2] = 0.0;
+  // TODO before merge (dhood): restore defaults and use intial pose topic subscription
+  init_pose_[0] = 20.896;
+  init_pose_[1] = 17.826;
+  init_pose_[2] = 2.49;
   init_cov_[0] = 0.5 * 0.5;
   init_cov_[1] = 0.5 * 0.5;
   init_cov_[2] = (M_PI/12.0) * (M_PI/12.0);
   // Check for NAN on input from param server, #5239
   double tmp_pos;
-  private_nh_.param("initial_pose_x", tmp_pos, init_pose_[0]);
+  node->get_parameter_or("initial_pose_x", tmp_pos, init_pose_[0]);
   if(!std::isnan(tmp_pos))
     init_pose_[0] = tmp_pos;
   else 
     ROS_WARN("ignoring NAN in initial pose X position");
-  private_nh_.param("initial_pose_y", tmp_pos, init_pose_[1]);
+  ROS_DEBUG("initial pose x: %.3f", init_pose_[0]);
+  node->get_parameter_or("initial_pose_y", tmp_pos, init_pose_[1]);
   if(!std::isnan(tmp_pos))
     init_pose_[1] = tmp_pos;
   else
     ROS_WARN("ignoring NAN in initial pose Y position");
-  private_nh_.param("initial_pose_a", tmp_pos, init_pose_[2]);
+  ROS_DEBUG("initial pose y: %.3f", init_pose_[1]);
+  node->get_parameter_or("initial_pose_a", tmp_pos, init_pose_[2]);
   if(!std::isnan(tmp_pos))
     init_pose_[2] = tmp_pos;
   else
     ROS_WARN("ignoring NAN in initial pose Yaw");
-  private_nh_.param("initial_cov_xx", tmp_pos, init_cov_[0]);
+  ROS_DEBUG("initial pose a: %.3f", init_pose_[2]);
+  node->get_parameter_or("initial_cov_xx", tmp_pos, init_cov_[0]);
   if(!std::isnan(tmp_pos))
     init_cov_[0] =tmp_pos;
   else
     ROS_WARN("ignoring NAN in initial covariance XX");
-  private_nh_.param("initial_cov_yy", tmp_pos, init_cov_[1]);
+  node->get_parameter_or("initial_cov_yy", tmp_pos, init_cov_[1]);
   if(!std::isnan(tmp_pos))
     init_cov_[1] = tmp_pos;
   else
     ROS_WARN("ignoring NAN in initial covariance YY");
-  private_nh_.param("initial_cov_aa", tmp_pos, init_cov_[2]);
+  node->get_parameter_or("initial_cov_aa", tmp_pos, init_cov_[2]);
   if(!std::isnan(tmp_pos))
     init_cov_[2] = tmp_pos;
   else
-    ROS_WARN("ignoring NAN in initial covariance AA");	
+    ROS_WARN("ignoring NAN in initial covariance AA");
 }
 
 void 
-AmclNode::checkLaserReceived(const ros::TimerEvent& event)
+AmclNode::checkLaserReceived()
 {
-  ros::Duration d = ros::Time::now() - last_laser_received_ts_;
+  // TODO before merge: fix timepoint subtraction
```

Comment:
Should this todo be resolved? The whole method is commented out. It's probably ok for the demo though.